modules/tinymce/src/core/main/ts/api/commands/ClipboardCommands.ts

```diff
@@ -1,37 +1,94 @@
-  editor.editorCommands.addCommands({
-    'Cut,Copy,Paste': (command) => {
-      const doc = editor.getDoc();
-      let failed;
-
-      // Try executing the native command
-      try {
-        doc.execCommand(command);
-      } catch (ex) {
-        // Command failed
-        failed = true;
-      }
 
-      // Chrome reports the paste command as supported however older IE:s will return false for cut/paste
-      if (command === 'paste' && !doc.queryCommandEnabled(command)) {
-        failed = true;
-      }
+  const copyToClipboard = async (text: string) => {
+    try {
+      await navigator.clipboard.writeText(text);
+      console.log('copyToClipboard', text);
+    } catch (err) {
+      handleClipboardError();
+    }
+  };
 
-      // Present alert message about clipboard access not being available
-      if (failed || !doc.queryCommandSupported(command)) {
-        let msg = editor.translate(
-          `Your browser doesn't support direct access to the clipboard. ` +
-          'Please use the Ctrl+X/C/V keyboard shortcuts instead.'
-        );
+  const pasteFromClipboard = async () => {
+    let text = '';
 
-        if (Env.os.isMacOS() || Env.os.isiOS()) {
-          msg = msg.replace(/Ctrl\+/g, '\u2318+');
-        }
+    try {
+      text = await navigator.clipboard.readText();
+      console.log('pasteFromClipboard', text);
+      editor.selection.setContent(text);
+    } catch (err) {
+      handleClipboardError();
+    }
+    return text;
+  };
 
-        editor.notificationManager.open({ text: msg, type: 'error' });
+  const cutToClipboard = async (text: string) => {
+    try {
+      await navigator.clipboard.writeText(text);
+      console.log('cutToClipboard', text);
+      editor.selection.setContent('');
+    } catch (err) {
+      handleClipboardError();
+    }
+  };
+
+  const fallbackClipboardCommand = (editor: Editor, command: string) => {
+    const doc = editor.getDoc();
+    let failed = false;
+
+    try {
+      doc.execCommand(command);
+    } catch (ex) {
+      failed = true;
+    }
+
+    // Chrome reports the paste command as supported, however older IE:s will return false for cut/paste
+    if (command === 'paste' && !doc.queryCommandEnabled(command)) {
+      failed = true;
+    }
+
+    if (failed || !doc.queryCommandSupported(command)) {
+      handleClipboardError();
+    }
+  };
+
+  const handleClipboardError = () => {
+
+    let msg = editor.translate(
+      `Your browser doesn't support direct access to the clipboard. ` +
+        'Please use the Ctrl+X/C/V keyboard shortcuts instead.'
+    );
+
+    if (Env.os.isMacOS() || Env.os.isiOS()) {
+      msg = msg.replace(/Ctrl\+/g, '\u2318+');
+    }
+
+    editor.notificationManager.open({ text: msg, type: 'error' });
+  };
+
+  editor.editorCommands.addCommands({
+    'Cut,Copy,Paste': (command) => {
+      const clipboard = navigator.clipboard;
+      console.log('command', navigator.clipboard);
```

Comment:
Please add a [secure context check](https://developer.mozilla.org/en-US/docs/Web/API/Window/isSecureContext) before poking the async clipboard API to avoid fatal errors in the console.